src/views/auth/auth.js

```diff
@@ -0,0 +1,27 @@
+import React, { PropTypes, Component } from 'react';
+
+export default class Auth extends Component {
+
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        loggedIn: PropTypes.bool.isRequired,
+        checkAuth: PropTypes.func.isRequired,
+        redirect: PropTypes.string.isRequired,
+        history: PropTypes.object.isRequired,
+    };
+
+    componentDidMount() {
+        const { loggedIn, checkAuth, redirect, history } = this.props;
+        if (!loggedIn) {
+            checkAuth(redirect, history);
+        }
+    }
+
+    render() {
+        const { loggedIn } = this.props;
+        if (loggedIn) {
+            return this.props.children;
```

Comment:
1. `componentDidMount` runs after render, but I can change it to `componentWillMount` instead. This will run before `render`.
2. We will only hit this endpoint once per session. So if you switch between components, we already set `loggedIn` to true, so we don't need to keep checking. If you refresh, it will reset the state, so we need to hit the endpoint. 

3. I'm about to push some changes that include this. It doesn't handle token expiration yet but it is something we'll need to handle